src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        throw Error('Something went wrong');
```

Comment:
I wouldn't throw an error here, because this way you expect that every consumer of this methods handles Errors. 
You could just: 

```suggestion
        console.error(`Error: ${action.type} is incorrect action type`);
```
We typically throw in `async` code, that will be taught later in the course. 

It's not inherently bad to throw, but there is no need for it here.
Also, `throwing` / `console.error` are not for the user, but for other programmers (or you) you should describe the problem that happened (In this case, bad action.type was provided)

Reply:
ESLint disallow to use of console, but you have a great point